apps/myaccount/src/components/shared/mobile-update-wizard-v2/mobile-update-wizard-v2.tsx

```diff
@@ -0,0 +1,262 @@
+import { setActiveForm } from "../../../store/actions/global";
+
+interface MobileUpdateWizardV2PropsInterface extends IdentifiableComponentInterface {
+    initialValue: string;
+    isOpen: boolean;
+    onClose: (isRevalidate?: boolean) => void;
+    onCancel: (isRevalidate?: boolean) => void;
+    initialStep?: number;
+    isMultiValued?: boolean;
+}
+
+const MobileUpdateWizardV2: FunctionComponent<MobileUpdateWizardV2PropsInterface> = (
+    {
+        initialValue,
+        isOpen,
+        onClose,
+        onCancel,
+        initialStep = 0,
+        isMultiValued = false,
+        ["data-componentid"]: testId = "component-id"
+    }: MobileUpdateWizardV2PropsInterface
+): ReactElement => {
+    const { t } = useTranslation();
+    const dispatch: Dispatch<any> = useDispatch();
+
+    const [ activeStep, setActiveStep ] = useState<number>(initialStep);
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+    const [ isOTPVerificationError, setIsOTPVerificationError ] = useState<boolean>(false);
+    const [ isOTPResendSuccess, setIsOTPResendSuccess ] = useState<boolean>(false);
+
+    const [ triggerUpdateMobile, setTriggerUpdateMobile ] = useTrigger();
+    const [ triggerVerifyOTP, setTriggerVerifyOTP ] = useTrigger();
+
+    const handleMobileNumberUpdate = (updatedValue: string): void => {
+        setIsSubmitting(true);
+        const data: PatchOperationRequest<ProfilePatchOperationValue> = {
+            Operations: [],
+            schemas: [ "urn:ietf:params:scim:api:messages:2.0:PatchOp" ]
+        };
+
+        data.Operations.push({
+            op: "add",
+            value: {
+                [ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get("PHONE_NUMBERS")]: [
+                    { type: "mobile", value: updatedValue }
+                ]
+            }
+        });
+
+        data.Operations.push({
+            op: "replace",
+            value: {
+                [ProfileConstants.SCIM2_SYSTEM_USER_SCHEMA] : {
+                    [ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get("VERIFY_MOBILE")] : true
+                }
+            }
+        });
+
+        updateProfileInfo(data as unknown as Record<string, unknown>)
+            .then((response: AxiosResponse) => {
+                if (response.status === 200) {
+                    if (response?.data
+                        ?.[ProfileConstants.SCIM2_SYSTEM_USER_SCHEMA]
+                        ?.pendingMobileNumber === updatedValue) {
+                        // Navigate to the OTP verification step.
+                        setActiveStep((activeStep: number) => activeStep + 1);
+                    } else {
+                        onClose(true);
+                    }
+                }
+            })
+            .catch((error: IdentityAppsApiException) => {
+                dispatch(addAlert({
+                    description:
+                        error?.response?.detail ??
```

Comment:
Lets show out own error messages rather than API once.
Check other applicable places as well.

Reply:
This is an already available code segment, which is refactored. Will follow up via another PR.